src/main/java/org/bptlab/cepta/DataToDatabase.java

```diff
@@ -0,0 +1,159 @@
+  }
+  public void insert(Object dataSet)
+      throws NoSuchFieldException, IllegalAccessException {
+
+    // Connection to PostgreSQL DB
+    ConnectionPool<PostgreSQLConnection> connection;
+
+    ConnectionPoolConfigurationBuilder config = new ConnectionPoolConfigurationBuilder();
+    config.setUsername(USER);
+    config.setPassword(PASSWORD);
+    config.setHost(HOST);
+    config.setPort(PORT);
+    config.setDatabase(DATABASE);
+    config.setMaxActiveConnections(100);
+    connection = PostgreSQLConnectionBuilder.createConnectionPool(config);
+
+    // stores strings of values and columns for sql query
+    String[] colsAndVals = columnsAndValuesToString(dataSet);
+
+    // Create query
+    String query = "INSERT INTO " + table_name + colsAndVals[0]
+        + " VALUES " + colsAndVals[1] + ";";
+    System.out.println(query);
+
+    // send query
+    CompletableFuture<QueryResult> future = connection.sendPreparedStatement(query);
+
+    // really execute query and get query result
+    try{
+      QueryResult result = future.get();
+    }catch (InterruptedException | ExecutionException e){
+      System.out.println("Could not get result");
+    }
+
+    // Close the connection pool
+    try{
+      connection.disconnect().get();
+    }catch (ExecutionException | InterruptedException e){
+      System.out.println("Could not disconnect");
+    }
+  }
+
+  private String[] columnsAndValuesToString(Object dataSet)
+      throws NoSuchFieldException, IllegalAccessException {
+    // takes the values matching to columns and converts them to
+    // a (val1, val2, ...) String
+    // necessary for usage in the sql statement
+
+    String valString = "(";
+    Class c = dataSet.getClass();
+
+    // get attribute fields of dataSet's class and store them in fields
+    Field[] fields = c.getDeclaredFields();
+
+    // stores names of attributes
+    String[] columns = new String[fields.length];
+/*
+    String[] columns = new String[fields.length];
+    for(int i = 0; i < fields.length; ++i){
+      columns[i] = fields[i].getName();
+    }
+*/
+
+    for(int i = 0; i < fields.length-1; i++){
+      // get attribute-field of class for column-name
+      Field f = fields[i];
+      f.setAccessible(true);
+
+      // add object's value of attribute/column to string
+      // with , in between
+      try{
+        if(f.getType().equals(" ".getClass())){
+          // add ' ' around value if it's a string
+          valString += "'" + f.get(dataSet).toString() + "'" + ", ";
+        }else{
+          valString += f.get(dataSet).toString() + ", ";
+        }
+        columns[i] = f.getName();
+      }catch (NullPointerException e){
+        /*
+         just go on with the for loop because there is no value (it's null)
+         we want to add to the database and so we don't need he column and
+         the value in our sql statement
+        */
+      }
+    }
+
+    // do the same for the last element but with ) instead of ,
```

Comment:
I feel like adding the parantheses in a single go would better since we don't split up the "we need to put parantheses around this string"-part into two places. 

We could try using String.format for that maybe.

Reply:
Yes, I think that is a lot nicer!